tests/python/rest_api/test_tasks.py

```diff
@@ -2684,3 +2687,46 @@ def test_can_export_and_import_skeleton_tracks_in_coco_format(self):
                 for tes in te.shapes
             ]
         )
+
+
+@pytest.mark.usefixtures("restore_db_per_function")
+@pytest.mark.usefixtures("restore_cvat_data")
+@pytest.mark.usefixtures("restore_redis_db_per_function")
+class TestTaskCreationOrder:
+    _USERNAME = "admin1"
+
+    def _create_task(self, idx: int) -> int:
+        task_spec = {
+            "name": f"Test task {idx}",
+            "labels": [
+                {
+                    "name": "car",
+                }
+            ],
+        }
+
+        task_data = {
+            "image_quality": 75,
+            "client_files": generate_image_files(3),
+        }
+
+        task_id, _ = create_task(self._USERNAME, task_spec, task_data)
+        return task_id
+
+    def test_can_create_tasks_in_well_formed_order(self):
```

Comment:
This test doesn't look reliable because of unpredictable execution order, and now we also don't test the case with disabled ordering. Why we can't add tests for both configurations in the server unit tests?

Reply:
Because server unit tests are not executed on the real instance? Generally, we don't need to test the order, we need to check that 1. some rq jobs depend on others, 2. There are no rq jobs that depend on several rq jobs. And looks like, these checks are predictable. Do you think it would be better if we keep server configuration with the disabled order in the public repo and move that test to private?

Reply:
> Generally, we don't need to test the order

If so, the test name should probably be changed.

> 1. some rq jobs depend on others, 2. ...

Probably, it's better to check that only 1 job has no deps, and all the others have 1 dep each. Otherwise we can get some unnoticed errors. Actually, the point 2 feels like an implementation detail.

> Do you think it would be better if we keep server configuration with the disabled order in the public repo and move that test to private?

No, I don't think so. If this feature is available in the public version, it should probably be tested in the public version. I'm worried that we currently test only 1 case.